week1/prep-exercises/1-counter/src/components/Counter.js

```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react';
+import './Counter.css';
+import Count from "./Count";
+import Button from './Button';
+
+
+function Counter() {
+    const [count, setCount] = useState(0)
+
+    let feedback = count > 10 ? "It's higher than 10!" : "Keep counting..."
+
+    function increment(step) {
+        setCount(prev => prev + step)
+    }
+
+    function decrement(step) {
+        if (count - step > 0) setCount(prev => prev - step) 
+        else setCount(0)
+    }
```

Comment:
we can make a call for this if it's difficult to understand 
you can compain these functions together in one function and we just through `step` as minus for decrementing
it's better also to put this function inside the `Button` function to make the Button and give it anything you want from this component then the Button component will be like this
```js
function Button({ step, count, setCount }) {
  const buttonName = step > 0 ? `increment ${step}` : `decrement ${step * -1}`;
  function increment(step) {
    if (step < 0 && count + step <= 0) setCount(0);
    else setCount((prev) => prev + step);
  }
  return (
    <button className="Counter-btn" onClick={() => increment(step)}>
      {buttonName}
    </button>
  );
}
```
 
and then you can call this component from `Counter` component like that
```js
<Button count={count} setCount={setCount} step={1} />
<Button count={count} setCount={setCount} step={-2} />
```